lib/galaxy/util/__init__.py

```diff
@@ -294,7 +295,7 @@ def unique_id(KEY_SIZE=128):
     return md5(random_bits).hexdigest()
 
 
-def parse_xml(fname: StrPath, strip_whitespace=True, remove_comments=True) -> ElementTree:
+def parse_xml(fname: Union[StrPath, Traversable], strip_whitespace=True, remove_comments=True) -> ElementTree:
```

Comment:
I am not sure that we want `Traversable` here, since for example we call `str(fname)` at line 306 below but the `Traversable` protocol doesn't have a `__str__()` method defined.

Maybe this is better:

```diff
diff --git a/test/unit/tool_util/data/test_tool_data_bundles.py b/test/unit/tool_util/data/test_tool_data_bundles.py
index 2e1a1f2717..9c6afddc57 100644
--- a/test/unit/tool_util/data/test_tool_data_bundles.py
+++ b/test/unit/tool_util/data/test_tool_data_bundles.py
@@ -15,7 +15,10 @@ from galaxy.util import (
     galaxy_directory,
     parse_xml,
 )
-from galaxy.util.resources import resource_path
+from galaxy.util.resources import (
+    as_file,
+    resource_path,
+)
 
 TOOLS_DIRECTORY = os.path.abspath(os.path.join(galaxy_directory(), "test/functional/tools/"))
 
@@ -51,8 +54,8 @@ def test_xml_parsing() -> None:
 
 
 def test_parsing_manual() -> None:
-    path = resource_path(__package__, "example_data_managers/manual.xml")
-    tree = parse_xml(path)
+    with as_file(resource_path(__package__, "example_data_managers/manual.xml")) as path:
+        tree = parse_xml(path)
     data_managers_el = tree.getroot()
     data_manager_el = data_managers_el.find("data_manager")
     description = convert_data_tables_xml(data_manager_el)
@@ -61,8 +64,8 @@ def test_parsing_manual() -> None:
 
 
 def test_parsing_mothur() -> None:
-    path = resource_path(__package__, "example_data_managers/mothur.xml")
-    tree = parse_xml(path)
+    with as_file(resource_path(__package__, "example_data_managers/mothur.xml")) as path:
+        tree = parse_xml(path)
     data_managers_el = tree.getroot()
     data_manager_el = data_managers_el.find("data_manager")
     description = convert_data_tables_xml(data_manager_el)
```

What do you think?

Reply:
Good catch; I've applied this